### PR TITLE
feat: change member role [AR-1603]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1603" title="AR-1603" target="_blank">StoryAR-1603</a>  Promote group convo members to admins (and demote them)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We need to be able to change the member's role in the group as an admin.

### Solutions

Created bottom sheet with role options to choose, added "edit" button on other user's profile group info under the current role which is available only for the admin, updated way of handling and showing snackbar messages on this screen.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Open member details as an admin of the group to change the role of such user in this group.

### Attachments (Optional)

https://user-images.githubusercontent.com/30429749/182564008-6248f826-e085-4647-813f-0d0b98bae91a.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-823/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4054950)(4054950)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4054950)(4054950)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-823/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4055444)(4055444)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4055444)(4055444)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt [hunk 1]
```diff
@@ -59,11 +66,14 @@ class OtherUserProfileScreenViewModel @Inject constructor(
...
-    qualifiedIdMapper: QualifiedIdMapper
+    qualifiedIdMapper: QualifiedIdMapper,
+    private val updateMemberRole: UpdateConversationMemberRoleUseCase
 ) : ViewModel() {
 
     var state: OtherUserProfileState by mutableStateOf(OtherUserProfileState())
-    var connectionOperationState: ConnectionOperationState? by mutableStateOf(null)
+
+    private val _infoMessage = MutableSharedFlow<UIText>()
+    val infoMessage = _infoMessage.asSharedFlow()
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt:76]:
```suggestion
    val infoMessage = infoMessage.asSharedFlow()
```

**Review by MohamadJaara — APPROVED** [review]
LGTM, only one naming issue 🚀

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt:76]: But how? 😄  There is `private val _infoMessage` and `val infoMessage` - the first with the underscore is mutable one, but we don't want to share mutable one, that's why there is a second one - public non-mutable

Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt:76]: ohhhhh my bad i did read it as
```
    val _infoMessage = infoMessage.asSharedFlow()
```

**Pull request merged**